akka-distributed-data/src/main/scala/akka/cluster/ddata/ReplicatedData.scala

```diff
@@ -37,6 +37,17 @@ trait ReplicatedData {
 
 }
 
+trait DeltaReplicatedData extends ReplicatedData {
+  type D <: ReplicatedData
+
+  def delta: Option[D]
+
+  def mergeDelta(d: D): T
+
```

Comment:
yes, it was because of `D` vs `T`, so it can be removed if we use `T` for the delta